Sources/SwiftParser/StringLiterals.swift

```diff
@@ -487,7 +487,7 @@ extension Parser {
       // string literal.
       guard currentToken.leadingTriviaText.isEmpty else { break }
 
-      if let stringSegment = self.consume(if: .stringSegment) {
+      if let stringSegment = self.consume(if: .stringSegment, .identifier) {
```

Comment:
This is not correct. `StringLiteralSegmentsSyntax` declares that the string segment must have a token kind `stringSegment` 

https://github.com/apple/swift-syntax/blob/d3c4f2461b75f05ad76a76d2c44da4e1a2350172/CodeGeneration/Sources/SyntaxSupport/ExprNodes.swift#L1399

This will fail if RawSyntax validation is enabled (which it is in CI).

https://github.com/apple/swift-syntax/blob/6bb19efac5ae49ec2de7d0b1d040b8e1e0a76de2/Package.swift#L27

I think it should be fine if you remapped the token to the `stringSegment` token kind.

Reply:
So I should consume and remap to `stringSegment `?

Reply:
Yes, that’s what I’d do